### PR TITLE
out_opentelemetry: add gzip compression

### PR DESCRIPTION
**Pull request by flobz**
Add GZIP compression option to opentelemetry output plugin. 

**Testing**
Before we can approve your change; please submit the following in a comment:
- [X] Example configuration file for the change
- [N/A] Debug log output from testing the change

- [X] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**

- [X] Documentation required for this feature

https://github.com/fluent/fluent-bit-docs/pull/939

**Backporting**

- [X] Backport to latest stable release.

### CODE REVIEW

**Comment by flobz** [comment]
Docs: https://github.com/fluent/fluent-bit-docs/pull/939
Valgrind test:

**Comment by flobz** [comment]
I have rebased this branch on a commit witch successfully pass the CI. Can you restart the CI workflow ?

**Comment by flobz** [comment]
Can someone restart the workflow please ? There was a timeout on a job.

File: plugins/out_opentelemetry/opentelemetry.c [hunk 1]
```diff
@@ -107,7 +122,9 @@ static int http_post(struct opentelemetry_context *ctx,
                             key->str, flb_sds_len(key->str),
                             val->str, flb_sds_len(val->str));
     }
-
+    if (compressed == FLB_TRUE) {
+        flb_http_set_content_encoding_gzip(c);
+    }
     ret = flb_http_do(c, &b_sent);
     if (ret == 0) {
```
Comment (edsiper) [inline plugins/out_opentelemetry/opentelemetry.c:129]: Your valgrind report says there is a memory leak:

```
==407985== LEAK SUMMARY:
==407985==    definitely lost: 925,638 bytes in 21 blocks
```

looking at the changes it seems like the new buffer allocated for the compressed gzip is not being free, so on every flush with gzip, you are leaking the allocated memory.

**Review by edsiper — CHANGES_REQUESTED** [review]

File: plugins/out_opentelemetry/opentelemetry.c (hunk 1 above)
Comment (flobz) [reply plugins/out_opentelemetry/opentelemetry.c:129]: ok thanks for the help. It should be fixed.

**Comment by edsiper** [comment]
thanks!

**Pull request merged**